internal/services/compute/restore_point_collection_resource.go

```diff
@@ -0,0 +1,211 @@
+	"github.com/hashicorp/go-azure-sdk/resource-manager/compute/2024-03-01/restorepointcollections"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/sdk"
+	"github.com/hashicorp/terraform-provider-azurerm/internal/tf/pluginsdk"
+)
+
+type RestorePointCollectionResource struct{}
+
+var _ sdk.ResourceWithUpdate = RestorePointCollectionResource{}
+
+func (r RestorePointCollectionResource) ModelObject() interface{} {
+	return &RestorePointCollectionResourceModel{}
+}
+
+type RestorePointCollectionResourceModel struct {
+	Name                   string                 `tfschema:"name"`
+	ResourceGroup          string                 `tfschema:"resource_group_name"`
+	Location               string                 `tfschema:"location"`
+	SourceVirtualMachineId string                 `tfschema:"source_virtual_machine_id"`
+	Tags                   map[string]interface{} `tfschema:"tags"`
+}
+
+func (r RestorePointCollectionResource) IDValidationFunc() pluginsdk.SchemaValidateFunc {
+	return restorepointcollections.ValidateRestorePointCollectionID
+}
+
+func (r RestorePointCollectionResource) ResourceType() string {
+	return "azurerm_restore_point_collection"
+}
+
+func (r RestorePointCollectionResource) Arguments() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{
+		"name": {
+			ForceNew: true,
+			Required: true,
+			Type:     pluginsdk.TypeString,
+		},
+
+		"resource_group_name": commonschema.ResourceGroupName(),
+
+		"location": commonschema.Location(),
+
+		"source_virtual_machine_id": {
+			Type:         pluginsdk.TypeString,
+			Required:     true,
+			ForceNew:     true,
+			ValidateFunc: commonids.ValidateVirtualMachineID,
+		},
+
+		"tags": commonschema.Tags(),
+	}
+}
+
+func (r RestorePointCollectionResource) Attributes() map[string]*pluginsdk.Schema {
+	return map[string]*pluginsdk.Schema{}
+}
+
+func (r RestorePointCollectionResource) Create() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 30 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Compute.RestorePointCollectionsClient
+			subscriptionId := metadata.Client.Account.SubscriptionId
+
+			var config RestorePointCollectionResourceModel
+			if err := metadata.Decode(&config); err != nil {
+				return fmt.Errorf("decoding: %+v", err)
+			}
+
+			id := restorepointcollections.NewRestorePointCollectionID(subscriptionId, config.ResourceGroup, config.Name)
+
+			existing, err := client.Get(ctx, id, restorepointcollections.DefaultGetOperationOptions())
+			if err != nil {
+				if !response.WasNotFound(existing.HttpResponse) {
+					return fmt.Errorf("checking for the presence of an existing %s: %+v", id, err)
+				}
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			parameters := restorepointcollections.RestorePointCollection{
+				Location: config.Location,
```

Comment:
```suggestion
				Location: location.Normalize(config.Location),
```
should we normalize the location here?